From 1ed8450126db7267e3d1b12e7ca90da8f7185d61 Mon Sep 17 00:00:00 2001
From: Fei Yang <fei.yang@intel.com>
Date: Wed, 22 May 2019 16:26:39 -0700
Subject: [PATCH 1/2] Revert "usb: gadget: f_fs: don't free buffer prematurely"

This reverts commit 9f8dc24f7f5d18d002b4f44bfc220770a421907f.

This patch fixed the kernel panic introduced by
    commit 772a7a724f69d258025fedd87dde1aafe4171aef
    usb: gadget: f_fs: Allow scatter-gather buffers

However, the scatter-gather buffers patch also caused adb push issue
where large file transfer fails. We still need to revert the sg buffer
patch for Android kernel.

Signed-off-by: Fei Yang <fei.yang@intel.com>
---
 drivers/usb/gadget/function/f_fs.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/function/f_fs.c b/drivers/usb/gadget/function/f_fs.c
index 47be961f1bf3..20413c276c61 100644
--- a/drivers/usb/gadget/function/f_fs.c
+++ b/drivers/usb/gadget/function/f_fs.c
@@ -1133,8 +1133,7 @@ static ssize_t ffs_epfile_io(struct file *file, struct ffs_io_data *io_data)
 error_mutex:
 	mutex_unlock(&epfile->mutex);
 error:
-	if (ret != -EIOCBQUEUED) /* don't free if there is iocb queued */
-		ffs_free_buffer(io_data);
+	ffs_free_buffer(io_data);
 	return ret;
 }
 
-- 
2.17.1

